client/components/data/query-post-counts/index.jsx

```diff
@@ -21,7 +21,9 @@ export default function QueryPostCounts( { siteId, type } ) {
 	const dispatch = useDispatch();
 
 	useEffect( () => {
-		dispatch( request( siteId, type ) );
+		if ( siteId ) {
```

Comment:
Let's guard against type too

Reply:
We can then drop the isRequired on props

Reply:
Done.